### PR TITLE
Feature/add logging

### PR DESCRIPTION
**Pull request by moonso**
- Fixed update case
- Added some logging

### CODE REVIEW

**Comment by moonso** [comment]
I've updated the changes from #64

File: scout/log/log.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    
+    Arguments:
+    
+        filename (str): Path to a file. Or None if logging is to
+                         be disabled.
+        loglevel (str): Determines the level of the log output.
+    """
+    
+    formatter = logging.Formatter(
+                    '[%(asctime)s] %(levelname)s: %(name)s: %(message)s'
```
Comment (robinandeer) [inline scout/log/log.py]: I just learned something new about formatting strings from this link: https://docs.python.org/3/howto/logging-cookbook.html#logging-to-multiple-destinations

Basically you can align each section of the log record so it's easier to read:

```python
formatter = logging.Formatter( 
    '[%(asctime)s] %(levelname)-8s: %(name)-25s: %(message)s')
```

File: scout/log/log.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import os
+import sys
+import logging
+
+def init_log(logger, filename=None, loglevel=None):
```
Comment (robinandeer) [inline scout/log/log.py]: I'm not super sure but why not initialize the root logger in this function? Then all the rest of the loggers will inherit it's formatting and handler, no?

```python
root_logger = logging.getLogger()
```

File: scout/log/log.py (hunk 1 above)
Comment (moonso) [inline scout/log/log.py]: Yes that is nice!

File: scout/log/log.py (hunk 2 above)
Comment (moonso) [inline scout/log/log.py]: Haha we just discussed it yeserday, remember?

Comment (moonso) [inline scout/log/log.py]: We concluded that it should allways be up to the user how the loggers should be formated.
So if you import the package you set the loggers yourself.

Comment (robinandeer) [inline scout/log/log.py]: Yeah :sweat_smile:, I'm still a little confused about this whole implicit inheritance works and what really is a root logger to some other type of logger, hehe. But what you say make sense - keep up the good work!

Comment (moonso) [inline scout/log/log.py]: Merge it when you feel ready.

Comment (robinandeer) [inline scout/log/log.py]: Sure thing - I think we can update the formatted template line first though :smiley:

**Pull request merged**